docs/UpgradeGuide.md

```diff
@@ -0,0 +1,568 @@
+| Configuration | Removed  | `Datadog.configure(client, options)`                                                                                 | Use `Datadog::Tracing.configure_onto(client, options)` instead.                                                                                                           |
+| Configuration | Removed  | `DD_#{integration}_ANALYTICS_ENABLED` and `DD_#{integration}_ANALYTICS_SAMPLE_RATE` environment variables            | Use `DD_TRACE_#{integration}_ANALYTICS_ENABLED` and `DD_TRACE_#{integration}_ANALYTICS_SAMPLE_RATE` instead                                                               |
+| Configuration | Removed  | `DD_PROPAGATION_INJECT_STYLE` and `DD_PROPAGATION_EXTRACT_STYLE` environment variables                               | Use `DD_PROPAGATION_STYLE_INJECT` and `DD_PROPAGATION_STYLE_EXTRACT` instead                                                                                              |
+| Configuration | Removed  | Setting a hash on `c.tracer.transport_options`                                                                       | Use a proc instead to pass the options to the desired adapter                                                                                                             |
+| Configuration | Removed  | Unused profiler configuration `c.profiling.exporter.transport_options`.                                              | Not supported.                                                                                                                                                            |
+| Integrations  | Changed  | `-` in HTTP header tag names are kept, and no longer replaced with `_`.                                              | For example: `http.response.headers.content_type` is changed to `http.response.headers.content-type`.                                                                     |
+| Integrations  | Changed  | `Contrib::Configurable#default_configuration` moved to `Tracing::Contrib::Configurable#new_configuration`.           | Use `Tracing::Contrib::Configurable#new_configuration` instead.                                                                                                           |
+| Integrations  | Changed  | `Datadog.configuration.registry` moved to `Datadog.registry`.                                                        | Use `Datadog.registry` instead.                                                                                                                                           |
+| Integrations  | Changed  | `Datadog::Ext::Integration` to `Datadog::Ext::Metadata`                                                              | Update your references.                                                                                                                                                   |
```

Comment:
Minor: A few of the entries end on a `.`, and a few don't. Perhaps worth making sure it's all consistent?